dio/lib/src/options.dart

```diff
@@ -659,8 +675,8 @@ class _RequestConfig {
 
   set headers(Map<String, dynamic>? headers) {
     _headers = caseInsensitiveKeyMap(headers);
-    if (_defaultContentType != null &&
-        !_headers.containsKey(Headers.contentTypeHeader)) {
+    if (!_headers.containsKey(Headers.contentTypeHeader) &&
+        _defaultContentType != null) {
```

Comment:
Why is there such a change?

Reply:
Thought that checking the headers first might gain a slightly better performance since it has a higher opportunity to have a content type than the options.